odoo/addons/base/models/ir_module.py

```diff
@@ -307,16 +308,33 @@ def unlink(self):
         self.clear_caches()
         return super(Module, self).unlink()
 
+    @staticmethod
+    def _check_python_external_dependency(pydep):
+        try:
+            pkg_resources.get_distribution(pydep)
+        except pkg_resources.DistributionNotFound as e:
+            try:
+                importlib.import_module(pydep)
+                _logger.warning("python external dependency %s should be replaced by it's PyPI package name", pydep)
+            except Exception:
```

Comment:
I would only catch `ImportError` and let other exceptions bubble up.

Reply:
@KangOl done